objects/src/transaction/proven_tx.rs

```diff
@@ -159,6 +125,292 @@ impl ProvenTransaction {
     }
 }
 
+// PROVEN TRANSACTION BUILDER
+// ================================================================================================
+
+/// Builder for a proven transaction.
+#[derive(Clone, Debug)]
+pub struct ProvenTransactionBuilder {
```

Comment:
I'm not sure we need such an elaborate builder - it seems to add quite a bit of code for relatively marginal benefit, IMO.

I think the main reason for introducing the builder is that `ProvenTransaction` constructor gets a bit unwieldy with too many arguments. There are some alternatives to consider:

1. Have a simpler builder. Fields like `account_id`, `initial_account_hash`, `final_account_hash`, `block_ref`, and `proof` are always required for a transaction (and there are no defaults for these fields). So, we could change the builder constructor to take them as arguments and then only optional fields would be added via methods.
2. Change the `ProvenTransaction` itself into a "builder" - i.e., make the constructor take only the required fields, and add input/output notes etc. via mutator methods.
3. Group some of the related fields into structs. For example, `account_id`, `initial_account_hash`, `final_account_hash`, and `account_details` could be grouped into a separate structs which would be provided to the constructor as a single entity.

I probably prefer the simpler builder approach as it seems cleaner (i.e., would reduce the number of potential runtime errors while simplifying the code).